<!DOCTYPE html><html lang="en"><head><meta charSet="utf-8"/><meta http-equiv="X-UA-Compatible" content="IE=edge"/><title>ChIP-seq-macs · Kaige Dong&#x27;s Site</title><meta name="viewport" content="width=device-width"/><meta name="generator" content="Docusaurus"/><meta name="description" content="from MACS manual ：***Identifying ChIP-seq enrichment using MACS***"/><meta name="docsearch:language" content="en"/><meta property="og:title" content="ChIP-seq-macs · Kaige Dong&#x27;s Site"/><meta property="og:type" content="website"/><meta property="og:url" content="https://kaigedong.github.io/Linux-best-practice/blog/2017/09/22/ChIP-seq-masc"/><meta property="og:description" content="from MACS manual ：***Identifying ChIP-seq enrichment using MACS***"/><meta property="og:image" content="https://kaigedong.github.io/Linux-best-practice/img/undraw_online.svg"/><meta name="twitter:card" content="summary"/><meta name="twitter:image" content="https://kaigedong.github.io/Linux-best-practice/img/undraw_tweetstorm.svg"/><link rel="shortcut icon" href="/Linux-best-practice/img/favicon.ico"/><link rel="stylesheet" href="//cdnjs.cloudflare.com/ajax/libs/highlight.js/9.12.0/styles/default.min.css"/><link rel="alternate" type="application/atom+xml" href="https://kaigedong.github.io/Linux-best-practice/blog/atom.xml" title="Kaige Dong&#x27;s Site Blog ATOM Feed"/><link rel="alternate" type="application/rss+xml" href="https://kaigedong.github.io/Linux-best-practice/blog/feed.xml" title="Kaige Dong&#x27;s Site Blog RSS Feed"/><script type="text/javascript" src="https://buttons.github.io/buttons.js"></script><script src="/Linux-best-practice/js/scrollSpy.js"></script><link rel="stylesheet" href="/Linux-best-practice/css/main.css"/><script src="/Linux-best-practice/js/codetabs.js"></script></head><body class="sideNavVisible separateOnPageNav"><div class="fixedHeaderContainer"><div class="headerWrapper wrapper"><header><a href="/Linux-best-practice/"><img class="logo" src="/Linux-best-practice/img/favicon.ico" alt="Kaige Dong&#x27;s Site"/><h2 class="headerTitleWithLogo">Kaige Dong&#x27;s Site</h2></a><a href="/Linux-best-practice/versions"><h3>0.1.5</h3></a><div class="navigationWrapper navigationSlider"><nav class="slidingNav"><ul class="nav-site nav-site-internal"><li class=""><a href="/Linux-best-practice/docs/doc1" target="_self">Docs</a></li><li class=""><a href="/Linux-best-practice/docs/doc4" target="_self">API</a></li><li class=""><a href="/Linux-best-practice/help" target="_self">Help</a></li><li class="siteNavGroupActive"><a href="/Linux-best-practice/blog/" target="_self">Blog</a></li></ul></nav></div></header></div></div><div class="navPusher"><div class="docMainWrapper wrapper"><div class="docsNavContainer" id="docsNav"><nav class="toc"><div class="toggleNav"><section class="navWrapper wrapper"><div class="navBreadcrumb wrapper"><div class="navToggle" id="navToggler"><div class="hamburger-menu"><div class="line1"></div><div class="line2"></div><div class="line3"></div></div></div><h2><i>›</i><span>All blog posts</span></h2><div class="tocToggler" id="tocToggler"><i class="icon-toc"></i></div></div><div class="navGroups"><div class="navGroup"><h3 class="navGroupCategoryTitle">All blog posts</h3><ul class=""><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2019/09/20/scrapy_to_search_engine_Introduction">1.爬虫课程的介绍</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2019/04/09/Python_sorting">Python排序算法</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2019/04/09/Learn_make_Donot_starve_mods">学习制作饥荒mod</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2019/04/09/Donot_starve_mod1">饥荒mod1</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2019/04/09/python_virtual_env">virtualenv 和 pyenv的使用</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/12/16/Wuenda_machine_learning">吴恩达机器学习笔记</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/12/13/Django-Apache-wsgi-_deploy">Django+Apache wsgi 部署</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/12/12/Apache-deploy-Django">Apache 部署 Django</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/12/09/pytorch-60min">PyTorch 60min教程</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/12/09/no_root_install_Apache">无root安装Apache</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/12/07/GIVEinstall">GIVE安装笔记</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/12/06/wali_v2ray">搬瓦工2: v2ray</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/08/16/Curses-python3-7">Curses_python3.7</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/05/12/wali_note">搬瓦工笔记1：ssr</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/19/DQN_5">强化学习实战：从零开始下五子棋</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/17/DQN_Sarsa">强化学习之Sarsa</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/17/DQN_Sarsa-lambda">增强学习之Sarsa(lambda)</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/16/DQN_Q-learning">深度学习02之Q-learning介绍</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/16/DQN">强化学习介绍与分类01</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/08/07_shadowsocks_settings">shadowsocks配置</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/07/19_shell_env_set/env-export">shell变量及进程及set,env,export</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/07/12_Perl/install_models">Perl安装模块与卸载</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/07/11_python_convert_exe_videos">python批量转换exe视频</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/12/09_no_root_install_MySQL">服务器无root安装MySQL</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/05/08_Macinstall_MySQL">Mac安装MySQL &amp; Ubuntu下安装MySQL记录</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/13/tmus">tmus指南</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/17_DQN_DQN">增强学习之DQN介绍</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/17_DQN2/Algo">增强学习之DQN算法</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/08/17python的decode/encode-与-r-b-u">python的decode/encode 与 r/b/u</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/08/17_golang_notes">golang笔记</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/12/09_unroot_install_UCSC_genome_browser">modele_name</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/08/13_Acer中Arch安装Secure/boot-EFI-设置">Acer中Arch安装Secure boot EFI 设置</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/13_HTMLandCSS_basic_03">HTML与CSS基础03-Web框架与表单设计</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2019/08/20_Linux_crontab">Linux crontab定时任务</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/13/learn-git">learn-git</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/13/items2">items2技巧</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/13/chenwei_gene_A07-A09">陈巍学基因A07-A09</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/13/chenwei_gene_A01-A03">陈巍学基因A01-A03</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/13/chenwei_geneA04-A06">陈巍学基因A04-A06</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/13/boxplot">数据可视化之柱状图</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/13/begin_building_sites">开始建站Hexo+NexT</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/13/Hexo-NexT404pages">Hexo-NexT主题的404页面</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/13/HTMLandCSS_basic_01">HTML和CSS基础01-html的语法和基本结构文档设置标记</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2018/04/13/HTML_CSS_basic2">HTML和CSS基础02-图像超链接和表格</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/26/pig_exons">modele_name</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/26/Counting-known-microRNAs-in-five-easy-steps">Counting known microRNAs in five easy steps</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/26/R_214">R极客的情人节</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/25/Adapter-and-quality-trimming-of-illumina-data">Adapter and quality trimming of illumina data</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/23/pytorch_bach">pytorch批训练</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/23/pytorch-Optimizer">pytorch:Optimizer优化器</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/23/pytorch-CNN">pytorch:CNN卷积神经网络</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/sync_server_files">同步服务器文件夹</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/pytorch07">pytorch07</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/pytorch06">pytorch06</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/pytorch05">pytorch05</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/Affymetrix">Affymetrix芯片原理</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/pytorch03">pytorch03</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/pytorch02">pytorch02</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/pytorch01">pytorch 学习笔记</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/matlab04">matlab04图像的增强处理</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/matlab03">matlab03图像的几何变换</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/matlab02">matlab02图像的点运算</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/matlab">matlab学习笔记01</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/WEB_frame">WEB框架</a></li><li class="navListItem navListItemActive"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/ChIP-seq-masc">ChIP-seq-macs</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/CSS">CSS层叠样式表</a></li><li class="navListItem"><a class="navItem" href="/Linux-best-practice/blog/2017/09/22/pytorch04">pytorch04</a></li></ul></div></div></section></div><script>
            var coll = document.getElementsByClassName('collapsible');
            var checkActiveCategory = true;
            for (var i = 0; i < coll.length; i++) {
              var links = coll[i].nextElementSibling.getElementsByTagName('*');
              if (checkActiveCategory){
                for (var j = 0; j < links.length; j++) {
                  if (links[j].classList.contains('navListItemActive')){
                    coll[i].nextElementSibling.classList.toggle('hide');
                    coll[i].childNodes[1].classList.toggle('rotate');
                    checkActiveCategory = false;
                    break;
                  }
                }
              }

              coll[i].addEventListener('click', function() {
                var arrow = this.childNodes[1];
                arrow.classList.toggle('rotate');
                var content = this.nextElementSibling;
                content.classList.toggle('hide');
              });
            }

            document.addEventListener('DOMContentLoaded', function() {
              createToggler('#navToggler', '#docsNav', 'docsSliderActive');
              createToggler('#tocToggler', 'body', 'tocActive');

              var headings = document.querySelector('.toc-headings');
              headings && headings.addEventListener('click', function(event) {
                var el = event.target;
                while(el !== headings){
                  if (el.tagName === 'A') {
                    document.body.classList.remove('tocActive');
                    break;
                  } else{
                    el = el.parentNode;
                  }
                }
              }, false);

              function createToggler(togglerSelector, targetSelector, className) {
                var toggler = document.querySelector(togglerSelector);
                var target = document.querySelector(targetSelector);

                if (!toggler) {
                  return;
                }

                toggler.onclick = function(event) {
                  event.preventDefault();

                  target.classList.toggle(className);
                };
              }
            });
        </script></nav></div><div class="container mainContainer postContainer blogContainer"><div class="wrapper"><div class="lonePost"><div class="post"><header class="postHeader"><h1 class="postHeaderTitle"><a href="/Linux-best-practice/blog/2017/09/22/ChIP-seq-masc">ChIP-seq-macs</a></h1><p class="post-meta">September 22, 2017</p><div class="authorBlock"><p class="post-authorName"><a href="" target="_blank" rel="noreferrer noopener">woobamboo</a></p></div></header><div><span><p>from MACS manual ：<strong><em>Identifying ChIP-seq enrichment using MACS</em></strong></p>
<p>步骤：<code>版本1.4</code></p>
<ol>
<li>安装macs</li>
<li>安装可选软件</li>
</ol>
<!--truncate-->
<pre><code class="hljs"><span class="hljs-keyword">bowtie; </span>samtools<span class="hljs-comment">;R;PeakSplitter;igv;</span>
</code></pre>
<pre><code class="hljs">下载<span class="hljs-keyword">bowtie </span>index<span class="hljs-comment">;</span>
</code></pre>
<ol start="3">
<li>运行MACS to call peaks</li>
</ol>
<ul>
<li><strong>以下用不同的数据来阐述如何使用不同的参数来运行macs。</strong></li>
</ul>
<p>用<strong>A来call FoxA1 peaks</strong>; 用<strong>B来 call H3K4me3 peaks 并打开片段大小评估</strong>；用<strong>C来call H3K4me3 peaks 用一个固定的片段大小</strong>；用<strong>D来call H3K4me4 peaks</strong></p>
<hr>
<h2><a class="anchor" aria-hidden="true" id="a-call-foxa1-peaks"></a><a href="#a-call-foxa1-peaks" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>A: Call FoxA1 peaks</h2>
<ul>
<li>解压hg19 index 文件：<code>unzip hg19.ebwt.zip</code></li>
<li>下载FoxA1 raw reads from <a href="http://cistrome.dfci.harvard.edu/MACSNatureProtocol/HAIB_T47D_FoxA1.tar.gz">http://cistrome.dfci.harvard.edu/MACSNatureProtocol/HAIB_T47D_FoxA1.tar.gz</a> 解压并将raw reads map到参考基因组上，使用bowtie：</li>
</ul>
<pre><code class="hljs css language-shell">tar xzvf HAIB_T47D_FoxA1.tar.gz
bowtie -m 1 -S -q /path_to/hg19 HAIB_T47D_FoxA1.fastq HAIB_T47D_FoxA1.sam
</code></pre>
<p><strong>参数说明</strong></p>
<blockquote>
<p><strong><code>-m 1</code></strong> #specifies that reads with only one hit on the genome are retained;</p>
<p><strong><code>-S</code></strong> #specifies the output format as SAM;</p>
<p><strong><code>-q</code></strong> #specifies the input format as FASTQTQ;</p>
<p><strong><code>/path_to/</code></strong> #is the directory containing the unzipped bowtie prebuilt indexes; and</p>
<p><strong><code>HAIB_T47D_FoxA1.fastq</code></strong> #contains the downloaded raw reads for FoxA1.</p>
</blockquote>
<ul>
<li>运行macs</li>
</ul>
<pre><code class="hljs css language-shell">macs -t HAIB_T47D_FoxA1.sam -n HAIB_T47D_FoxA1 -g hs -B -S --call-subpeaks
</code></pre>
<p><strong>MACS版本</strong></p>
<blockquote>
<p>直接使用<code>macs</code>调用的是<code>1.4.2</code>
要使用macs 2 可使用命令：<code>macs2</code></p>
</blockquote>
<h3><a class="anchor" aria-hidden="true" id="输出文件说明"></a><a href="#输出文件说明" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>输出文件说明</h3>
<table>
<thead>
<tr><th>File name</th><th>Description</th></tr>
</thead>
<tbody>
<tr><td>HAIB_T47D_FoxA1_model.r</td><td>一个R脚本可用来生成峰的PDF</td></tr>
<tr><td>HAIB_T47D_FoxA1_model.pdf</td><td>执行R脚本后生成，PDF文件描述了read在峰上的分布和片段大小评估</td></tr>
<tr><td>HAIB_T47D_FoxA1_peaks.xls</td><td>MACS使用的主要参数和鉴定出来的峰的细节信息</td></tr>
<tr><td>HAIB_T47D_FoxA1_peaks.bed</td><td>峰的坐标BED格式</td></tr>
<tr><td>HAIB_T47D_FoxA1_peaks.subpeaks.bed</td><td>Subpeak locations in BED-like format.由PeakSplitter 生成, 被MACS调用。</td></tr>
<tr><td>HAIB_T47D_FoxA1_summits.bed</td><td>Summit locations of the peaks in BED format</td></tr>
<tr><td>HAIB_T47D_FoxA1_MACS_bedGraph</td><td>Directory where the BedGraph files are generated. For each control or ChIP-seq sample, a BedGraph file describes the read distribution along the whole genome</td></tr>
</tbody>
</table>
<p><strong>参数说明</strong></p>
<blockquote>
<p><strong><code>-t</code></strong>  指定ChIP-seq read比对结果的文件名。MACS支持并能自动识别： SAM, BAM, BED, ELAND, ELANDMULTI, ELANDMULTIPET, ELANDEXPORT and BOWTIE. 使用参数<code>--format</code> 可以覆盖自动识别的格式。</p>
<p><strong><code>-g</code></strong> 指定基因组大小。<code>hs</code> 参数是人基因组大小的近似的缩写。等于2.7e9.</p>
<p><strong><code>-n</code></strong> 应用<code>‘HAIB_T47D_FoxA1’</code>作为输出文件的前缀。</p>
<p><strong><code>-B</code></strong>  生成bedGraph格式的文件包含拓展的每个碱基对的read 堆积的信号文件。这个步骤很耗费内存和时间。因此，只有在需要bedGraph文件时，才指定-B.</p>
<p><strong><code>-S</code></strong>对于真个基因组生成单一的bedGraph 文件。否则的话，会对每个染色体生成一个。</p>
</blockquote>
<p><strong>其他可能会遇到的参数</strong></p>
<blockquote>
<p><strong><code>--bw</code></strong> sets the ‘bandwidth’, which is half of the sliding window size used in the model-building step.</p>
<p><strong><code>--mfold</code></strong> specifies an interval of high-confidence enrichment ratio against the background on which to build the model. The default value ‘10, 30’ means that a model will be built on the basis of regions having read counts that are 10- to 30-fold of the background.</p>
<p><strong><code>--pvalue</code></strong> establishes a threshold P value: only peaks surpassing the threshold will be reported. The default threshold is 10 − 5.Users can first set a loose P value cutoff so that a sufficient number of peaks will be reported and then select peaks having the smallest P values for downstream analyses.</p>
</blockquote>
<ul>
<li>检查MACS在终端的输出状态：</li>
</ul>
<pre><code class="hljs"><span class="hljs-builtin-name">INFO</span> @ Sun, 03 Jun 2012 23:36:03:
<span class="hljs-comment"># ARGUMENTS LIST:</span>
<span class="hljs-comment"># name  =   HAIB_T47D_FoxA1</span>
<span class="hljs-comment"># format  =   AUTO</span>
<span class="hljs-comment"># ChIP-seq file  =   HAIB_T47D_FoxA1.sam</span>
<span class="hljs-comment"># control file  =   None</span>
<span class="hljs-comment"># effective genome size  =   2.70e + 09</span>
<span class="hljs-comment"># band width  =   300</span>
<span class="hljs-comment"># model fold  =   10,30</span>
<span class="hljs-comment"># pvalue cutoff  =   1.00e-05</span>
<span class="hljs-comment"># Large dataset will be scaled towards smaller dataset.</span>
<span class="hljs-comment"># Range for calculating regional lambda is: 10000 bps</span>
<span class="hljs-builtin-name">INFO</span> #1 read tag files.
<span class="hljs-builtin-name">INFO</span> #1 read treatment tags.
<span class="hljs-builtin-name">INFO</span> Detected format is: SAM
 &lt; Several lines are skipped &gt; 
<span class="hljs-builtin-name">INFO</span> #2 Build Peak Model.
<span class="hljs-builtin-name">INFO</span> #2 number of paired peaks: 16586
<span class="hljs-builtin-name">INFO</span> #2 finished!
<span class="hljs-builtin-name">INFO</span> #2 predicted fragment length is 114 bps
<span class="hljs-builtin-name">INFO</span> #2.2 Generate R<span class="hljs-built_in"> script </span><span class="hljs-keyword">for</span> model : HAIB_T47D_FoxA1_model.r
<span class="hljs-builtin-name">INFO</span> #3 Call peaks.
<span class="hljs-builtin-name">INFO</span> #3 shift treatment data
<span class="hljs-builtin-name">INFO</span> #3 merge  + /- strand of treatment data
<span class="hljs-builtin-name">INFO</span> #3 save the shifted <span class="hljs-keyword">and</span> merged tag counts into bedGraph file.
<span class="hljs-builtin-name">INFO</span> write <span class="hljs-keyword">to</span> a bedGraph file
 &lt; Several lines are skipped &gt; 
<span class="hljs-builtin-name">INFO</span> #3 call peak candidates
<span class="hljs-builtin-name">INFO</span> #3 use self <span class="hljs-keyword">to</span> calculate local lambda <span class="hljs-keyword">and</span><span class="hljs-built_in"> filter </span>peak candidates.
<span class="hljs-builtin-name">INFO</span> #3 Finally, 74761 peaks are called!
<span class="hljs-builtin-name">INFO</span> #4 Write output xls file. HAIB_T47D_FoxA1_peaks.xls
<span class="hljs-builtin-name">INFO</span> #4 Write peak bed file. HAIB_T47D_FoxA1_peaks.bed
<span class="hljs-builtin-name">INFO</span> #4 Write summits bed file. HAIB_T47D_FoxA1_summits.bed
<span class="hljs-builtin-name">INFO</span> #5 Done! Check the output files!
<span class="hljs-builtin-name">INFO</span> #6 Try <span class="hljs-keyword">to</span> invoke PeakSplitter.
<span class="hljs-builtin-name">INFO</span> #6 Please check HAIB_T47D_FoxA1_peaks.subpeaks.bed file <span class="hljs-keyword">for</span> PeakSplitter 
output!
</code></pre>
<blockquote>
<p>输出的信息如下：<strong>日期（第一行）；主要参数（以<code>#</code>开头的行）；运行过程（以<code>INFO</code>开头的行）。</strong>运行过程中的时间信息已经被移除了（为了更直观）。如果MACS遇到了异常（比如MACS评估一个片段大小太小），警告信息将会出现。</p>
</blockquote>
<blockquote>
<p>! 注意 尽管警告信息不影响MACS运行，仍需要认真对待警告。例如，</p>
<p>警告信息：<strong><code>unbalanced reads between treatment and control</code></strong>means that the FDR of the resulting peaks will be overestimated when the control sample has more reads and will be underestimated when the ChIP-seq sample is sequenced more deeply.</p>
<p>警告信息：<strong><code>Fewer paired peaks X than 1,000</code></strong> means that MACS only identified X model peaks and may indicate potential data quality issues because 1,000 model peaks are needed to robustly estimate ChIP-DNA fragment size.</p>
<p>警告信息<strong><code>missing chromosome X data</code></strong>might suggest that <strong>the raw input file for that chromosome is incomplete</strong>.</p>
</blockquote>
<ul>
<li>生成peak模型的PDF图像（假设R已经被安装）</li>
</ul>
<pre><code class="hljs css language-R">Rscript HAIB_T47D_FoxA1_model.r
</code></pre>
<blockquote>
<p>在当前目录生成名为<code>HAIB_T47D_FoxA1_model.pdf</code>的文件，图像描述了reads在峰的模型中正义链反义链上的分布和评估的片段的大小。</p>
</blockquote>
<hr>
<h2><a class="anchor" aria-hidden="true" id="b-call-h3k4me3-peaks"></a><a href="#b-call-h3k4me3-peaks" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>B: call H3K4me3 peaks</h2>
<ul>
<li><strong>数据的获取</strong>：数据包含一个<code>control replicate</code> 和一个 <code>ChIP-seq replicate</code></li>
</ul>
<pre><code class="hljs css language-shell">nohup wget -c  http://cistrome.dfci.harvard.edu/MACSNatureProtocol/UW_K562_H3K4me3.tar.gz &amp;
tar xvzf UW_K562_H3K4me3.tar.gz
</code></pre>
<p><strong>运行MACS:</strong></p>
<pre><code class="hljs css language-shell">macs -t UW_K562_H3K4me3.bam -c UW_K562_H3K4me3_Control.bam -g hs -n UW_K562_H3K4me3 -B -S --call-subpeaks
</code></pre>
<blockquote>
<p>参数<code>-c</code>指明了control 样本的文件名。</p>
</blockquote>
<ul>
<li>**检查MACS的屏幕输出，如果运行成功，MACS会输出如下信息：</li>
</ul>
<pre><code class="hljs"> &lt; Several lines are skipped &gt; 
<span class="hljs-builtin-name">INFO</span> : #2 Build Peak Model.
<span class="hljs-builtin-name">INFO</span> : #2 number of paired peaks: 12267
<span class="hljs-builtin-name">INFO</span> : #2 finished!
<span class="hljs-builtin-name">INFO</span> : #2 predicted fragment length is 156 bps
 &lt; Several lines are skipped &gt; 
<span class="hljs-builtin-name">INFO</span> : #3 use control data <span class="hljs-keyword">to</span><span class="hljs-built_in"> filter </span>peak candidates.
<span class="hljs-builtin-name">INFO</span> : #3 Finally, 20632 peaks are called!
<span class="hljs-builtin-name">INFO</span> : #3 <span class="hljs-builtin-name">find</span> negative peaks by swapping treat <span class="hljs-keyword">and</span> control
<span class="hljs-builtin-name">INFO</span> : #3 Finally, 4006 peaks are called!
 &lt; Several lines are skipped &gt; 
</code></pre>
<ul>
<li><strong>生成PDF图像:</strong></li>
</ul>
<pre><code class="hljs css language-shell">Rscript UW_K562_H3K4me3_model.r
</code></pre>
<blockquote>
<p>对于输出文件：因为这一步比上一个多用了一个control sample，会生成额外的文件<code>UW_K562_H3K4me3_negative_peaks.xls</code>包含了peaks called by comparing the control sample with the ChIP-seq sample. 这些peaks 被MACS用来评估报告的ChIP-seq峰的FDR。</p>
</blockquote>
<hr>
<h2><a class="anchor" aria-hidden="true" id="c-call-h3k4me3-peaks-with-a-specified-dna-fragment-size"></a><a href="#c-call-h3k4me3-peaks-with-a-specified-dna-fragment-size" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>C: call H3K4me3 peaks with a specified DNA fragment size</h2>
<ul>
<li><strong>下载数据</strong>：包含一个ChIP-Seq replicate 和两个control replicates。<strong>两个control要被合并</strong>。</li>
</ul>
<pre><code class="hljs css language-shell">nohup wget -c  http://cistrome.dfci.harvard.edu/MACSNatureProtocol/BROAD_GM12878_H3K4me3.tar.gz 1&gt;download.log 2&gt;&amp;1&amp;
tar xvzf BROAD_GM12878_H3K4me3.tar.gz
samtools merge BROAD_GM12878_H3K4me3_Control.bam BROAD_GM12878_H3K4me3_Control_1.bam BROAD_GM12878_H3K4me3_Control_2.bam
</code></pre>
<ul>
<li><strong>运行MACS：</strong></li>
</ul>
<pre><code class="hljs css language-shell">macs -t BROAD_GM12878_H3K4me3.bam -c BROAD_GM12878_H3K4me3_Control.bam -g hs -n BROAD_GM12878_H3K4me3 -B -S --call-subpeaks
</code></pre>
<ul>
<li>检查屏幕输出，应该包含下列行：</li>
</ul>
<pre><code class="hljs"> &lt; Several lines are skipped &gt; 
INFO : #<span class="hljs-number">2</span> number of paired peaks: <span class="hljs-number">31077</span>
INFO : #<span class="hljs-number">2</span> finished!
INFO : #<span class="hljs-number">2</span> predicted fragment length <span class="hljs-keyword">is</span> <span class="hljs-number">53</span> bps
 &lt; Several lines are skipped &gt; 
</code></pre>
<blockquote>
<p>MACS 建立的模型片段长度为53，是不同寻常的短在典型的ChIP-Seq 实验中。因此，修改参数重新运行MACS：</p>
</blockquote>
<ul>
<li>可使用Ctrl + C 终止MACS的运行。然后移除文件夹：<code>BROAD_GM12878_H3K4me3_MACS_bedGraph</code></li>
</ul>
<pre><code class="hljs css language-shell">rm -rf BROAD_GM12878_H3K4me3_MACS_bedGraph
</code></pre>
<ul>
<li>重新运行MACS：</li>
</ul>
<pre><code class="hljs css language-shell">macs14 -t BROAD_GM12878_H3K4me3.bam -c BROAD_GM12878_H3K4me3_Control.bam -g hs -n BROAD_GM12878_H3K4me3 --nomodel --shiftsize 73 -B -S --call-subpeak
</code></pre>
<hr>
<h2><a class="anchor" aria-hidden="true" id="d-calling-h3k36me3-peaks"></a><a href="#d-calling-h3k36me3-peaks" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>D:Calling H3K36me3 peaks</h2>
<ul>
<li><strong>下载数据</strong>:该数据包含两个control replicates 和两个ChIP-seq replicates</li>
</ul>
<pre><code class="hljs css language-shell">  http://cistrome.dfci.harvard.edu/MACSNatureProtocol/BROAD_GM12878_H3K36me3.tar.gz
  tar xvzf BROAD_GM12878_H3K36me3.tar.gz
  samtools merge BROAD_GM12878_H3K36me3.bam BROAD_GM12878_H3K36me3_1.bam BROAD_GM12878_H3K36me3_2.bam
  samtools merge BROAD_GM12878_H3K36me3_Control.bam BROAD_GM12878_H3K36me3_Control_1.bam BROAD_GM12878_H3K36me3_Control_2.bam
</code></pre>
<ul>
<li><strong>Run MACS:</strong></li>
</ul>
<pre><code class="hljs">macs -t <span class="hljs-keyword">BROAD_GM12878_H3K36me3.bam </span>-c <span class="hljs-keyword">BROAD_GM12878_H3K36me3_Control.bam </span>-g hs -n <span class="hljs-keyword">BROAD_GM12878_H3K36me3 </span>--nomodel --<span class="hljs-keyword">shiftsize </span><span class="hljs-number">73</span> -<span class="hljs-keyword">B </span>-S --pvalue <span class="hljs-number">1</span>e-3 --call-<span class="hljs-keyword">subpeaks
</span></code></pre>
<blockquote>
<p>与默认<code>--pvalue 1e-5</code>相比，这一步使用的p-value<code>--pvalue 1e-3</code>不太严格。因为H3K36me3 ChIP-seq 数据经常来自有更宽的不太集中的区域，参数<code>--nomodel</code> 和 <code>--shiftsize 73</code>优先采用。</p>
</blockquote>
<hr>
<h1><a class="anchor" aria-hidden="true" id="使用igv加载macs的结果"></a><a href="#使用igv加载macs的结果" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>使用IGV加载MACS的结果</h1>
<ol>
<li>首先解压并重命名 <code>bedGraph</code> 文件。以FoxA1数据集为例，首先定位到<code>HAIB_T47D_FoxA1_MACS_bedGraph/treat</code>,</li>
<li>解压文件 <code>gzip -d HAIB_T47D_FoxA1_treat_afterfiting_all.bdg.gz</code></li>
<li>更改后缀为<code>bedGraph</code>，这样可被IGV识别<code>mv HAIB_T47D_FoxA1_treat_afterfiting_all.bdg HAIB_T47D_FoxA1_treat_afterfiting_all.bedGraph</code></li>
<li>用IGV加载<code>HAIB_T47D_FoxA1_treat_afterfiting_all.bedGraph</code>，根据IGV的手册。</li>
</ol>
<hr>
<h2><a class="anchor" aria-hidden="true" id="macs2的用法部分"></a><a href="#macs2的用法部分" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>macs2的用法(部分)</h2>
<p><a href="https://pypi.python.org/pypi/MACS2">https://pypi.python.org/pypi/MACS2</a></p>
<h3><a class="anchor" aria-hidden="true" id="usage-of-macs2"></a><a href="#usage-of-macs2" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Usage of MACS2</h3>
<pre><code class="hljs">macs2 <span class="hljs-string">[-h]</span> <span class="hljs-string">[--version]</span>
{callpeak,filterdup,bdgpeakcall,bdgcmp,randsample,bdgdiff,bdgbroadcall}
</code></pre>
<h4><a class="anchor" aria-hidden="true" id="example-for-regular-peak-calling"></a><a href="#example-for-regular-peak-calling" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Example for regular peak calling</h4>
<pre><code class="hljs">macs2 callpeak -t ChIP<span class="hljs-selector-class">.bam</span> -c Control<span class="hljs-selector-class">.bam</span> -f BAM -g hs -n test -B -<span class="hljs-selector-tag">q</span> <span class="hljs-number">0.01</span>
</code></pre>
<h3><a class="anchor" aria-hidden="true" id="example-for-broad-peak-calling"></a><a href="#example-for-broad-peak-calling" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>Example for broad peak calling</h3>
<pre><code class="hljs"><span class="hljs-symbol">macs2</span> callpeak -t ChIP.<span class="hljs-keyword">bam </span>-c Control.<span class="hljs-keyword">bam </span>--<span class="hljs-keyword">broad </span>-g hs --<span class="hljs-keyword">broad-cutoff </span><span class="hljs-number">0</span>.<span class="hljs-number">1</span>
</code></pre>
<h3><a class="anchor" aria-hidden="true" id="几个常用的-sub-commands"></a><a href="#几个常用的-sub-commands" aria-hidden="true" class="hash-link"><svg class="hash-link-icon" aria-hidden="true" height="16" version="1.1" viewBox="0 0 16 16" width="16"><path fill-rule="evenodd" d="M4 9h1v1H4c-1.5 0-3-1.69-3-3.5S2.55 3 4 3h4c1.45 0 3 1.69 3 3.5 0 1.41-.91 2.72-2 3.25V8.59c.58-.45 1-1.27 1-2.09C10 5.22 8.98 4 8 4H4c-.98 0-2 1.22-2 2.5S3 9 4 9zm9-3h-1v1h1c1 0 2 1.22 2 2.5S13.98 12 13 12H9c-.98 0-2-1.22-2-2.5 0-.83.42-1.64 1-2.09V6.25c-1.09.53-2 1.84-2 3.25C6 11.31 7.55 13 9 13h4c1.45 0 3-1.69 3-3.5S14.5 6 13 6z"></path></svg></a>几个常用的<strong>sub-commands</strong></h3>
<p><strong>callpeak</strong>:   Main MACS2 Function to Call peaks from alignment results.
<strong>bdgpeakcall</strong>:    Call peaks from bedGraph output.
<strong>bdgbroadcall</strong>:   Call broad peaks from bedGraph output.
<strong>bdgcmp</strong>: Deduct noise by comparing two signal tracks in bedGraph.
<strong>bdgdiff</strong>:    Differential peak detection based on paired four bedgraph files.
<strong>filterdup</strong>eads at the same position, then convert acceptable format to BED format.
<strong>predictd</strong>:   Predict d or fragment size from alignment results.
<strong>pileup</strong>: Pileup aligned reads with a given extension size (fragment size or d in MACS language). Note there will be no step for duplicate reads filtering or sequencing depth scaling, so you may need to do certain post- processing.
<strong>randsample</strong>: Randomly sample number/percentage of total reads.
<strong>refinepeak</strong>: (Experimental) Take raw reads alignment, refine peak summits and give scores measuring balance of forward- backward tags. Inspired by SPP.</p>
<p><strong>下面只讲<code>callpeak</code>模块</strong>，其他内容可以<code>macs2 COMMAND -h</code>来查看细节</p>
<ul>
<li><p><strong>Call peaks</strong></p>
<p>是macs2的主要功能，可以用<code>macs2 callpeak</code>命令直接调用。如果不加参数直接输这个命令，你将会看到这个参数的详细描述。</p></li>
<li><p><strong>Options</strong></p>
<ul>
<li><code>-t/–treatment FILENAME</code>:这是唯一必须的参数，如果你有多个alignment files，你可以这样指明它们<code>-t A B C</code>,macs 将会pool up all these files</li>
<li><code>-c/–control</code>：与<code>-t</code>参数相同</li>
<li><code>-n/–name</code>:实验的名字，MACS将会使用这个名字来生成输出文件如<code>NAME_peaks.xls</code></li>
<li><code>–outdir</code>:MACS将通过这个参数将文件输出到指定的文件夹</li>
<li><code>-f/–format FORMAT</code>:Format of tag file, can be “ELAND”, “BED”, “ELANDMULTI”, “ELANDEXPORT”, “ELANDMULTIPET” (for pair-end tags), “SAM”, “BAM”, “BOWTIE”, “BAMPE” or “BEDPE”. Default is “AUTO” which will allow MACS to decide the format automatically. “AUTO” is also usefule when you combine different formats of files. Note that MACS can’t detect “BAMPE” or “BEDPE” format with “AUTO”, and you have to implicitly specify the format for “BAMPE” and “BEDPE”.</li>
</ul></li>
</ul>
</span></div></div><div class="blogSocialSection"></div></div><div class="blog-recent"><a class="button" href="/Linux-best-practice/blog">Recent posts</a></div></div></div><nav class="onPageNav"><ul class="toc-headings"><li><a href="#a-call-foxa1-peaks">A: Call FoxA1 peaks</a><ul class="toc-headings"><li><a href="#输出文件说明">输出文件说明</a></li></ul></li><li><a href="#b-call-h3k4me3-peaks">B: call H3K4me3 peaks</a></li><li><a href="#c-call-h3k4me3-peaks-with-a-specified-dna-fragment-size">C: call H3K4me3 peaks with a specified DNA fragment size</a></li><li><a href="#d-calling-h3k36me3-peaks">D:Calling H3K36me3 peaks</a></li><li><a href="#macs2的用法部分">macs2的用法(部分)</a><ul class="toc-headings"><li><a href="#usage-of-macs2">Usage of MACS2</a></li><li><a href="#example-for-broad-peak-calling">Example for broad peak calling</a></li><li><a href="#几个常用的sub-commands">几个常用的<strong>sub-commands</strong></a></li></ul></li></ul></nav></div><footer class="nav-footer" id="footer"><section class="sitemap"><a href="/Linux-best-practice/" class="nav-home"><img src="/Linux-best-practice/img/favicon.ico" alt="Kaige Dong&#x27;s Site" width="66" height="58"/></a><div><h5>Docs</h5><a href="/Linux-best-practice/docs/en/doc1.html">Getting Started (or other categories)</a><a href="/Linux-best-practice/docs/en/doc2.html">Guides (or other categories)</a><a href="/Linux-best-practice/docs/en/doc3.html">API Reference (or other categories)</a></div><div><h5>Community</h5><a href="/Linux-best-practice/en/users.html">User Showcase</a><a href="https://stackoverflow.com/questions/tagged/" target="_blank" rel="noreferrer noopener">Stack Overflow</a><a href="https://discordapp.com/">Project Chat</a><a href="https://twitter.com/" target="_blank" rel="noreferrer noopener">Twitter</a></div><div><h5>More</h5><a href="/Linux-best-practice/blog">Blog</a><a href="https://github.com/">GitHub</a><a class="github-button" data-icon="octicon-star" data-count-href="/facebook/docusaurus/stargazers" data-show-count="true" data-count-aria-label="# stargazers on GitHub" aria-label="Star this project on GitHub">Star</a></div></section><a href="https://opensource.facebook.com/" target="_blank" rel="noreferrer noopener" class="fbOpenSource"><img src="/Linux-best-practice/img/oss_logo.png" alt="Facebook Open Source" width="170" height="45"/></a><section class="copyright">Copyright © 2019 Kaige Dong</section></footer></div></body></html>